src/pipeline/client.rs

```diff
@@ -356,19 +350,17 @@ where
 
     fn call(&mut self, req: Request) -> Self::Future {
         let (tx, rx) = tokio::sync::oneshot::channel();
-        #[cfg(feature = "tracing")]
         let span = tracing::Span::current();
-        #[cfg(not(feature = "tracing"))]
-        let span = ();
-        event!(span, Level::TRACE, "issuing request");
+        span.in_scope(|| tracing::event!(Level::TRACE, "issuing request"));
```

Comment:
could be:
```suggestion
        tracing::trace!(parent: &span, "issuing request");
```
or just
```suggestion
        tracing::trace!("issuing request");
```
as per my above comment about the expected behavior of `Span::current()`.